apps/miniapp/domains/common/utils/oidcAuth.tsx

```diff
@@ -40,7 +51,54 @@ const OidcAuthProvider = ({ children }) => {
             }
         }
 
-    }, [isAllLoaded, user, isAuthenticated, condoUser, condoOrganization])
+    }, [isAllLoaded, user, isAuthenticated, condoUser, condoOrganization, hasStorageAccess])
+
+    useEffect(() => {
+        if (!hasStorageAccess && document && document.hasStorageAccess) {
+            document.hasStorageAccess().then(
+                (hasAccess) => {
+                    // Boolean hasAccess says whether the document has access or not.
+                    console.debug(`document.hasStorageAccess() resolved: ${hasAccess}`)
+                    setHasStorageAccess(hasAccess)
+                },
+                (reason) => {
+                    // Promise was rejected for some reason.
+                    console.debug(`document.hasStorageAccess() rejected: ${reason}`)
+                    setHasStorageAccess(false)
+                },
+            )
+        }
+    }, [hasStorageAccess])
+
+    const requestAccessToStorage = useCallback(() => {
+        if (!hasStorageAccess && document && document.requestStorageAccess) {
+            document && document.requestStorageAccess().then(
```

Comment:
you've already checked `document` presence in previous row

Reply:
Fixed